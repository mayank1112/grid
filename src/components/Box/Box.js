import React from 'react';
import PropTypes from 'prop-types';

import './box.css';

const Box = ({ children, style, onMouseOver, onMouseOut, onClick }) => (
    <div
        className='box'
        style={style}
        onClick={onClick}
        onMouseOut={onMouseOut}
        onMouseOver={onMouseOver}
    >{children}
    </div>
);

Box.propTypes = {
    style: PropTypes.object.isRequired,
    children: PropTypes.string.isRequired,
    onMouseOver: PropTypes.func.isRequired,
    onMouseOut: PropTypes.func.isRequired,
    onClick: PropTypes.func.isRequired,
};

export default Box;
