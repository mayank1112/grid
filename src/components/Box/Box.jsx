import React from 'react';
import PropTypes from 'prop-types';

import './Box.css';

const Box = ({ children, onClick }) => (
    <div
        className='box'
        onClick={onClick}
    >{children}</div>
);

Box.propTypes = {
    children: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
};

export default Box;
