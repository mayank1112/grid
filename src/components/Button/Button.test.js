import Enzyme, { shallow } from 'enzyme';
import Box from './Button.jsx';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

it('<Box> component should match snapshot', () => {
    const component = shallow(<Box style={{}} onClick={() => { }} onMouseOver={() => { }} onMouseOut={() => { }}>1</Box>);
    expect(toJson(component)).toMatchSnapshot();
});
