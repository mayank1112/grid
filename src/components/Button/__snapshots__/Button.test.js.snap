// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box> component should match snapshot 1`] = `
<div
  className="box"
  onClick={[Function]}
>
  1
</div>
`;
